components/serialization/json/src/main/java/com/microsoft/kiota/serialization/JsonParseNode.java

```diff
@@ -208,10 +209,46 @@ private <T> List<T> iterateOnArray(Function<JsonParseNode, T> fn) {
     @Nonnull public <T extends Parsable> T getObjectValue(@Nonnull final ParsableFactory<T> factory) {
         Objects.requireNonNull(factory, "parameter factory cannot be null");
         final T item = factory.create(this);
+        if (item.getClass() == UntypedNode.class) return (T) getUntypedValue();
         assignFieldValues(item, item.getFieldDeserializers());
         return item;
     }
 
+    @Nonnull private UntypedNode getUntypedValue() {
+        return getUntypedValue(currentNode);
+    }
+
+    @Nonnull private UntypedNode getUntypedValue(JsonElement element) {
+        if (element.isJsonNull()) return new UntypedNull();
+        else if (element.isJsonPrimitive()) {
+            final JsonPrimitive primitive = element.getAsJsonPrimitive();
+            if (primitive.isBoolean()) return new UntypedBoolean(primitive.getAsBoolean());
+            else if (primitive.isString()) return new UntypedString(primitive.getAsString());
+            else if (primitive.isNumber()) return new UntypedDouble(primitive.getAsDouble());
```

Comment:
when are we effectively using demical/float/long/integer? I know it can be hard to tell without the requested type and solely from from the data.

Reply:
I believe I did this because of the challenge you speak of. A similar implementation observed in the `tryGetAnything()` method at https://github.com/microsoft/kiota-java/blob/41bafdbc61d0ee8cc36de993874e4bd793024dda/components/serialization/json/src/main/java/com/microsoft/kiota/serialization/JsonParseNode.java#L276
I ended up keeping the other types as they can still be used during serialization though.

Reply:
The only thing I'm a bit worried about is the symmetrical aspect of it.
If somebody serializes as int, they'd probably expect to get an int back.
But I guess as long as this choice is consistent across languages, we can re-evaluate later based on customer feedback.